File: Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = max(nums)
        Sum = 0
        for i in range(len(nums)):
            Sum += nums[i]
            ans = max(ans,Sum)
            if(Sum < 0):
                Sum = 0
        return ans

            

        

File: Maximum subarray.java

class Solution {
    public int maxProduct(int[] nums) {
        int mx = Integer.MIN_VALUE;
        int mul = 1;
        for(int i=0;i<nums.length;i++){
            mul *= nums[i];
            mx = Math.max(mx,mul);
            if(mul == 0){
                mul = 1;
            }
        }
        mul = 1;
        for(int i=nums.length-1;i>=0;i--){
            mul *= nums[i];
            mx = Math.max(mx,mul);
            if(mul == 0){
                mul = 1;
            }
        }
        return mx;

        
    }
}
